cli: Panic in `cockroach sql` with URL

From @donpdonp on gitter:

```
donp@commons:~$ COCKROACH_URL=postgresql://db1:26257?sslmode=disable cockroach sql
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x68c196]

goroutine 1 [running]:
panic(0x17096c0, 0xc420012090)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/pkg/cli.preparePrompts(0xc42001000e, 0x26, 0x1787fc0, 0x0, 0x61, 0x18)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/sql.go:358 +0x1d6
github.com/cockroachdb/cockroach/pkg/cli.(*cliState).doStart(0xc420438320, 0x2, 0xc420438320)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/sql.go:398 +0xdb
github.com/cockroachdb/cockroach/pkg/cli.runInteractive(0xc420426cf0, 0xc420422400, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/sql.go:781 +0x31b
github.com/cockroachdb/cockroach/pkg/cli.runTerm(0x2316920, 0x26d30f8, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/sql.go:859 +0x224
github.com/cockroachdb/cockroach/pkg/cli.maybeDecorateGRPCError.func1(0x2316920, 0x26d30f8, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/error.go:30 +0x5a
github.com/spf13/cobra.(*Command).execute(0x2316920, 0x26d30f8, 0x0, 0x0, 0x2316920, 0x26d30f8)
	/go/src/github.com/spf13/cobra/command.go:632 +0x23e
github.com/spf13/cobra.(*Command).ExecuteC(0x2312740, 0xc42000c200, 0x8c574b56306ae3d3, 0xc42000c238)
	/go/src/github.com/spf13/cobra/command.go:722 +0x367
```

The panic is because the URL is missing the `user@` component.

--- a/pkg/cli/interactive_tests/test_url_login.tcl
+++ b/pkg/cli/interactive_tests/test_url_login.tcl
@@ -1,0 +1,17 @@
+#! /usr/bin/env expect -f
+
+source [file join [file dirname $argv0] common.tcl]
+
+start_server $argv
+
+# Check that the client can start when no username is specified.
+# This is run as an acceptance test to ensure that the code path
+# that generates the interactive prompt presented to the user is
+# also exercised by the test.
+spawn $argv sql --url "postgresql://localhost:26257?sslmode=disable"
+eexpect @localhost
+
+send "\004"
+eexpect eof
+
+stop_server $argv

--- a/pkg/cli/sql.go
+++ b/pkg/cli/sql.go
@@ -353,9 +353,13 @@ func preparePrompts(dbURL string) (fullPrompt string, continuePrompt string) {
 	// continued statement prompt is: "        -> "
 	fullPrompt = dbURL
 	if parsedURL, err := url.Parse(dbURL); err == nil {
+		username := ""
+		if parsedURL.User != nil {
+			username = parsedURL.User.Username()
+		}
 		// If parsing fails, we keep the entire URL. The Open call succeeded, and that
 		// is the important part.
-		fullPrompt = fmt.Sprintf("%s@%s", parsedURL.User.Username(), parsedURL.Host)
+		fullPrompt = fmt.Sprintf("%s@%s", username, parsedURL.Host)
 	}
 
 	if len(fullPrompt) == 0 {

